core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -502,6 +522,8 @@ private[spark] class BlockManagerInfo(
 
   def blocks: JHashMap[BlockId, BlockStatus] = _blocks
 
+  def cachedBlocks: mutable.HashSet[BlockId] = _cachedBlocks
```

Comment:
also, the point of using a private map and a public accessor is that the caller cannot mutate the underlying map. I think it makes more sense to return an immutable set here